CMakeLists.txt

```diff
@@ -1,16 +1,58 @@
 cmake_minimum_required (VERSION 3.3)
 project (llama)
 
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(CMAKE_CXX_STANDARD 11)
+# llama
+find_package(Boost 1.66.0 REQUIRED)
+add_library(${PROJECT_NAME} INTERFACE)
+target_include_directories(${PROJECT_NAME} INTERFACE $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include> $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
+target_compile_features(${PROJECT_NAME} INTERFACE cxx_std_11)
+target_link_libraries(${PROJECT_NAME} INTERFACE Boost::headers)
 
-ADD_SUBDIRECTORY("examples")
+# llama::llama to make subdirectory projects work
+add_library(${PROJECT_NAME}::${PROJECT_NAME} ALIAS ${PROJECT_NAME})
 
+# llama IDE
+file(GLOB_RECURSE llamaSources "${CMAKE_CURRENT_SOURCE_DIR}/include/**")
+add_custom_target("llamaIde" SOURCES ${llamaSources})
```

Comment:
What is an IDE target?

Reply:
I do not how if the term is widly used. What it means is that header-only libraries usually have no buildable targets in CMake. So in IDEs (like CLion, VS, etc.), the header files composing the library do not show up. Therefore, I created a custom target, that is not build, but contains all the library header files so they are shown in an IDE.

Reply:
Okay, understandable. Well, I don't know the best practice here. I would probably suggest a name like `llamaHeaders` or `llamaHeaderOnly`, but I guess `llamaIDE` would also be fine if you would add a small comment above the definition in the `CmakeLists.txt` :)

Reply:
I took the same name as it's used in alpaka and mallocMC. I have no strong preference.

Reply:
Sounds good, then probably just an additional line of comment will clarify this :)

Reply:
Done